### PR TITLE
[App] Connect novu notifications to UI

### PR DESCRIPTION
**Pull request by rners01**
**Description:** Connect novu notifications to UI
**Notion Task:** https://www.notion.so/App-Connect-Alerts-UI-to-the-real-data-from-Novu-57267998bec6419aaf352d5d64942226
**Notion Project:** https://www.notion.so/saladtech/Frontend-Add-push-notifications-to-the-web-app-3a60520ff8d4485d85965b12fa1c2e84

### CODE REVIEW

File: packages/web-app/src/modules/home-views/components/NavigationBar.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  const warningsNotifications = bannerNotifications.filter((notification) => notification?.variant === 'error')
+  const hasUnseenNotifications = unseenCount > 0
+  const notifications: Pick<NavigationBarProps, 'notifications'> = {
+    ...props.notifications,
+    news: newsNotifications,
+    warnings: warningsNotifications,
+    hasUnseenNotifications
+  }
+
+  return <GardenNovigationBar {...props} notifications={notifications} />
```
Comment (vitto-moz) [inline packages/web-app/src/modules/home-views/components/NavigationBar.tsx:26]:
```suggestion
  return <GardenNavigationBar {...props} notifications={notifications} />
```

File: packages/web-app/src/modules/notifications/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    }
+  } catch (e) {
+    return null
+  }
+}
+
+interface NotificationActionData {
+  label: string
+  onClick: () => void
+}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/utils.tsx:77]:
```suggestion
interface NotificationAction {
  label: string
  onClick: () => void
}
```

File: packages/web-app/src/modules/notifications/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    return { value: Math.ceil(interval.minutes), unit: 'minute' }
+  }
+
+  return { value: interval.seconds ? Math.ceil(interval.seconds) : 0, unit: 'second' }
+}
+
+export const getConfiguredNovuBannerNotifications = (
+  novuNotifications: IMessage[] | undefined,
+  onAcknowledge: (notificationID: string) => void,
+  onDismiss: (notificationID: string) => void,
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/utils.tsx:107]:
```suggestion
  onAcknowledge: (notificationId: string) => void,
  onDismiss: (notificationId: string) => void,
```

File: packages/web-app/src/modules/notifications/models/NotificationMessage.tsx [hunk 4]
```diff
@@ -52,3 +54,60 @@ export interface NotificationMessage {
...
+      openLink: OpenLinkNotificationAction;
+  }
+}
+
+export interface NovuNotification {
+  /** The resource identifier. */
+  id: string;
+  /** The Novu resource identifier. */
+  novuId: string;
+  /** The title. */
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/models/NotificationMessage.tsx:95]: we used these 2 ids in widget to distinguish SaladBowl notifications and Novu notifications.
What is the purpose we have them here?

File: packages/web-app/src/modules/notifications/models/NotificationMessage.tsx [hunk 5]
```diff
@@ -52,3 +54,60 @@ export interface NotificationMessage {
...
+  title: string;
+  /** The body. */
+  body: string;
+  /** The date and time of the notification. */
+  createTime: Date | undefined;
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/models/NotificationMessage.tsx:100]:
```suggestion
  createDate: Date | undefined;
```

**Review by vitto-moz — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/home-views/components/NavigationBar.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { useNotifications } from '@novu/notification-center'
+import { NavigationBar as GardenNovigationBar, type NavigationBarProps } from '@saladtechnologies/garden-components'
+import type { FunctionComponent } from 'react'
+import { getConfiguredNovuBannerNotifications } from '../../notifications/utils'
+
+export const NavigationBar: FunctionComponent<NavigationBarProps> = (props) => {
```
Comment (vitto-moz) [inline packages/web-app/src/modules/home-views/components/NavigationBar.tsx:6]:
```suggestion
export const NavigationBarWithNotifications: FunctionComponent<NavigationBarProps> = (props) => {
```

File: packages/web-app/src/modules/home-views/components/NavigationBar.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { useNotifications } from '@novu/notification-center'
+import { NavigationBar as GardenNovigationBar, type NavigationBarProps } from '@saladtechnologies/garden-components'
```
Comment (vitto-moz) [inline packages/web-app/src/modules/home-views/components/NavigationBar.tsx:2]:
```suggestion
import { NavigationBar, type NavigationBarProps } from '@saladtechnologies/garden-components'
```

**Review by vitto-moz — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/notifications/models/NotificationMessage.tsx [hunk 8]
```diff
@@ -52,3 +54,58 @@ export interface NotificationMessage {
...
+/** A resource that represents an action that opens a link in the default browser. */
+export interface OpenLinkNotificationAction {
+  /** The title. */
+  title: string;
+  /** The link. */
+  link: string;
+}
+
+/** A resource that represents an in-app notification action. */
+export interface NovuNotificationAction {
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/models/NotificationMessage.tsx:77]: Seems the naming is misleading here
the interface below has nothing common with novu notification template 
```suggestion
export interface NotificationAction {
```

File: packages/web-app/src/modules/notifications/models/NotificationMessage.tsx [hunk 9]
```diff
@@ -52,3 +54,58 @@ export interface NotificationMessage {
...
+  } | {
+    $case: "dismiss";
+    dismiss: DismissNotificationAction;
+  } | {
+      $case: "openLink";
+      openLink: OpenLinkNotificationAction;
+  }
+}
+
+export interface NovuNotification {
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/models/NotificationMessage.tsx:90]: Seems the naming is misleading here: this interface describes the application needs, but not describe the content of novu notification
```suggestion
export interface Notification {
```

File: packages/web-app/src/modules/notifications/utils.tsx [hunk 10]
```diff
@@ -0,0 +1,180 @@
+import type { IMessage } from '@novu/shared'
+import type { NotificationBannerProps } from '@saladtechnologies/garden-components'
+import { DateTime } from 'luxon'
+import { NotificationMessageCategory, type NovuNotification, type NovuNotificationAction } from './models'
+
+interface NotificationCta {
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/utils.tsx:6]:
```suggestion
interface NovuTemplateCta {
```

File: packages/web-app/src/modules/notifications/utils.tsx [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+  acknowledgeLabel: string
+  linkCTA: {
+    linkLabel: string
+    url: string
+  }
+}
+
+type NotificationCtaContent = NotificationCta['acknowledgeLabel'] | NotificationCta['linkCTA']
```
Comment (vitto-moz) [inline packages/web-app/src/modules/notifications/utils.tsx:14]:
```suggestion
type NovuTemplateCtaContent = NotificationCta['acknowledgeLabel'] | NotificationCta['linkCTA']
```

**Review by vitto-moz — CHANGES_REQUESTED** [review]

**Review by Maks19 — CHANGES_REQUESTED** [review]

**Review by Maks19 — APPROVED** [review]

**Review by vitto-moz — DISMISSED** [review]

**Pull request closed without merging**